core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -378,7 +378,10 @@ where
         const MAX_TRANSACTION_CHECKS: usize = 10_000;
         let mut transaction_ids = Vec::with_capacity(MAX_TRANSACTION_CHECKS);
 
-        while let Some(id) = self.container.pop() {
+        while transaction_ids.len() < MAX_TRANSACTION_CHECKS {
+            let Some(id) = self.container.pop() else {
+                break;
```

Comment:
not sure why the fmt was different here than on master - needed to add a `;` in b7dc0e2cbb400cb19403167349ce8889561ff6f2